flatsurf/geometry/gl2r_orbit_closure.py

```diff
@@ -988,41 +989,28 @@ def decomposition(self, v, limit=-1):
             decomposition.decompose(int(limit))
         return Decomposition(self, decomposition, u)
 
-    def _visit_slope(self, visited, x, y):
-        if x == 0:
-            slope = 1e1337
-        else:
-            slope = float(y) / float(x)
-
-        if slope not in visited:
-            visited[slope] = []
+    def decompositions(self, bound, limit=-1, bfs=False):
+        limit = int(limit)
 
-        for a, b in visited[slope]:
-            if b * x == a * y: return False
+        connections = self._surface.connections().bound(int(bound))
+        if bfs:
+            connections = connections.byLength()
 
-        visited[slope].append((x, y))
+        Vector = cppyy.gbl.flatsurf.Vector[type(self._surface).Coordinate]
+        slopes = cppyy.gbl.std.set[Vector, Vector.CompareSlope]()
 
-        return True
+        for connection in connections:
+            direction = connection.vector()
+            if slopes.find(direction) != slopes.end():
```

Comment:
We cannot say `direction in slopes` due to https://bitbucket.org/wlav/cppyy/issues/278/membership-test-in-std-set-is-broken-slow.